presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestRedundantSortColumnsRemoval.java

```diff
@@ -0,0 +1,124 @@
+
+        // Negative test - No prefix matches the grouping set, so TopN columns are not pruned
+        tester().assertThat(ImmutableSet.of(new MergeLimitWithSort(), new RemoveRedundantTopNColumns()), logicalPropertiesProvider)
+                .on("SELECT orderkey, custkey, sum(totalprice), min(orderdate) FROM orders GROUP BY orderkey, custkey " +
+                        "ORDER BY orderkey, sum(totalprice), custkey, min(orderdate) LIMIT 10")
+                .doesNotMatch(topNMatchWith(ImmutableList.of(sort("ORDERKEY", ASCENDING, LAST), sort("CUSTKEY", ASCENDING, LAST))));
+    }
+
+    @Test
+    public void testRemoveRedundantColumnsFromSort()
+    {
+        // OrderBy prefix matches GroupBy columns clause exactly
+        tester().assertThat(ImmutableSet.of(new RemoveRedundantSortColumns()), logicalPropertiesProvider)
+                .on("SELECT orderkey, custkey, sum(totalprice), min(orderdate) FROM orders GROUP BY orderkey, custkey " +
+                        "ORDER BY orderkey, custkey, sum(totalprice), min(orderdate)")
+                .matches(sortWith(ImmutableList.of(sort("ORDERKEY", ASCENDING, LAST), sort("CUSTKEY", ASCENDING, LAST))));
+
+        // Flipped order matches too since the Grouping set remains the same
+        tester().assertThat(ImmutableSet.of(new RemoveRedundantSortColumns()), logicalPropertiesProvider)
+                .on("SELECT orderkey, custkey, sum(totalprice), min(orderdate) FROM orders GROUP BY orderkey, custkey " +
+                        "ORDER BY custkey, orderkey, sum(totalprice), min(orderdate)")
+                .matches(sortWith(ImmutableList.of(sort("CUSTKEY", ASCENDING, LAST), sort("ORDERKEY", ASCENDING, LAST))));
+
+        // No impact due to sort direction
+        tester().assertThat(ImmutableSet.of(new RemoveRedundantSortColumns()), logicalPropertiesProvider)
+                .on("SELECT orderkey, custkey, sum(totalprice), min(orderdate) FROM orders GROUP BY orderkey, custkey " +
+                        "ORDER BY custkey DESC, orderkey ASC, sum(totalprice), min(orderdate)")
+                .matches(sortWith(ImmutableList.of(sort("CUSTKEY", DESCENDING, LAST), sort("ORDERKEY", ASCENDING, LAST))));
+
+        // Negative test - No prefix matches the grouping set, so TopN columns are not pruned
+        tester().assertThat(ImmutableSet.of(new RemoveRedundantSortColumns()), logicalPropertiesProvider)
+                .on("SELECT orderkey, custkey, sum(totalprice), min(orderdate) FROM orders GROUP BY orderkey, custkey " +
+                        "ORDER BY orderkey, sum(totalprice), custkey, min(orderdate)")
+                .doesNotMatch(sortWith(ImmutableList.of(sort("ORDERKEY", ASCENDING, LAST), sort("CUSTKEY", ASCENDING, LAST))));
+    }
+
+    private static PlanMatchPattern topNMatchWith(ImmutableList<PlanMatchPattern.Ordering> orderBy)
+    {
+        return output(
+                topN(10, orderBy,
```

Comment:
nit: Implicit assumption everyone using this will always use `LIMIT 10`. Even if just used in this file, might be good to pass N as an argument?

Reply:
Was practising YAGNI and leaving it up for a later refactor